src/ms3/instrument_defaults.csv

```diff
@@ -1,652 +1,583 @@
-10 hole f diatonic harmonica,harmonica-d12f,,22,10 Hole F Diatonic Harmonica,Harm.,10 Hole F Diatonic Harmonica,wind.reed.harmonica,10 Hole F Diatonic Harmonica
-10 hole d diatonic harmonica,harmonica-d12d,,22,10 Hole D Diatonic Harmonica,Harm.,10 Hole D Diatonic Harmonica,wind.reed.harmonica,10 Hole D Diatonic Harmonica
-10 hole c diatonic harmonica,harmonica-d12c,,22,10 Hole C Diatonic Harmonica,Harm.,10 Hole C Diatonic Harmonica,wind.reed.harmonica,10 Hole C Diatonic Harmonica
-10 hole a diatonic harmonica,harmonica-d12a,,22,10 Hole A Diatonic Harmonica,Harm.,10 Hole A Diatonic Harmonica,wind.reed.harmonica,10 Hole A Diatonic Harmonica
-10 hole g diatonic harmonica,harmonica-d12-g,,22,10 Hole G Diatonic Harmonica,Harm.,10 Hole G Diatonic Harmonica,wind.reed.harmonica,10 Hole G Diatonic Harmonica
-10 hole low d diatonic harmonica,harmonica-d10low-d,,22,10 Hole Low D Diatonic Harmonica,Harm.,10 Hole Low D Diatonic Harmonica,wind.reed.harmonica,10 Hole Low D Diatonic Harmonica
-20 hole chordet harmonica,harmonica-chordet,,22,20 Hole Chordet Harmonica,Harm.,20 Hole Chordet Harmonica,wind.reed.harmonica,20 Hole Chordet Harmonica
-bass harmonica,bass-harmonica,,22,Bass Harmonica,Bs. Harm.,Bass Harmonica,wind.reed.harmonica.bass,Bass Harmonica
-bass hohner harmonica,bass-harmonica-hohner,,22,Bass Hohner Harmonica,Bs. Harm.,Bass Hohner Harmonica,wind.reed.harmonica.bass,Bass Hohner Harmonica
-bass huang harmonica,bass-harmonica-huang,,22,Bass Huang Harmonica,Bs. Harm.,Bass Huang Harmonica,wind.reed.harmonica.bass,Bass Huang Harmonica
-accordion,accordion,,21,Accordion,Acc.,Accordion,keyboard.accordion,Accordion
-accordion.1,accordion,,21,Accordion,Acc.,Accordion,keyboard.accordion,Accordion
-bandoneon,bandoneon,,23,Bandoneon,Ban.,Bandoneon,keyboard.bandoneon,Bandoneon
-bandoneon.1,bandoneon,,23,Bandoneon,Ban.,Bandoneon,keyboard.bandoneon,Bandoneon
-concertina,concertina,,21,Concertina,Conc.,Concertina,keyboard.concertina,Concertina
-concertina.1,concertina,,21,Concertina,Conc.,Concertina,keyboard.concertina,Concertina
-melodica,melodica,,22,Melodica,Mel.,Melodica,wind.reed.melodica,Melodica
-sheng,sheng,,20,Sheng,She.,Sheng,wind.reed.sheng,Sheng
-soprano sheng,soprano-sheng,,20,Soprano Sheng,S She.,Soprano Sheng,wind.reed.sheng,Soprano Sheng
-alto sheng,alto-sheng,,20,Alto Sheng,A She.,Alto Sheng,wind.reed.sheng,Alto Sheng
-tenor sheng,tenor-sheng,,20,Tenor Sheng,T She.,Tenor Sheng,wind.reed.sheng,Tenor Sheng
-bass sheng,bass-sheng,,20,Bass Sheng,B She.,Bass Sheng,wind.reed.sheng,Bass Sheng
-brass,brass,,61,Brass,Br.,Brass,brass.group,Brass
-brass.1,brass,,61,Brass,Br.,Brass,brass.group,Brass
+,id,ChannelName,ChannelValue,controllers,group,staff_type_name,keysig,defaultClef,longName,shortName,trackName,instrumentId,part_trackName,useDrumset,clef
+piccolo,piccolo,[None],[72],"[[{'ctrl': '32', 'value': '17'}]]",pitched,stdNormal,,,Piccolo,Picc.,Piccolo,wind.flutes.flute.piccolo,Piccolo,,
```

Comment:
Is there a particular reason why the second and third columns contain lists (`[None]`, `[72]`) and the third one a dictionary nested in two lists, the whole thing converted into a string? Not only does it not look good, it is especially very error-prone.

Please do not include lists if it can be avoided. For the controllers, you could simply create one column per `ctrl` (we could name them `ctrl_32`, for instance) and fill in values only where a particular control value is used.

Reply:
The reason is the channel tag structure: we need to keep the order of the channels and inside the list of controllers for each channel. If you change the values of the controllers in the wrong channel, the instrument will not play back correctly.